### PR TITLE
Use new Joint APIs for Parent/Child name

### PR DESCRIPTION
**Pull request by scpeters**
# 🦟 Bug fix

Follow-up to gazebosim/sdformat#1053.

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [X] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [X] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: src/EntityComponentManager_TEST.cc [hunk 1]
```diff
@@ -2874,9 +2874,9 @@ TEST_P(EntityComponentManagerFixture,
   manager.CreateComponent(jointEntity, components::Name("jointEntity"));
   manager.CreateComponent(jointEntity, components::Joint());
   manager.CreateComponent(jointEntity,
-      components::ParentLinkName(parentLinkEntityName));
+      components::ParentName(parentLinkEntityName));
```
Comment (azeey) [inline src/EntityComponentManager_TEST.cc:2877]: I see you're changing the component names here, but i don't see the corresponding changes in `components/ParentLinkName.hh` and `components/ChildLineName.hh`. I'm not sure if we've deprecated components before, but I think we can create a `ParentName`/`ChildName` component and alias it to `ParentLinkName`/`ChildLinkName` with a deprecation attribute.

**Review by azeey — CHANGES_REQUESTED** [review]

File: src/EntityComponentManager_TEST.cc (hunk 1 above)
Comment (scpeters) [reply src/EntityComponentManager_TEST.cc]: https://github.com/gazebosim/gz-sim/pull/1548/commits/41dd47ddc5a4a102346e19c995f04ca44625a8cf

**Review by chapulina — APPROVED** [review]
CI is unstable right now, we should retrigger it soon

**Comment by scpeters** [comment]
Ubuntu has a few test failures but no deprecation warnings

* [Build Status](https://build.osrfoundation.org/job/ignition_gazebo-ci-pr_any-ubuntu_auto-amd64/8968/) https://build.osrfoundation.org/job/ignition_gazebo-ci-pr_any-ubuntu_auto-amd64/8968/

**Review by azeey — APPROVED** [review]

**Pull request merged**